src/librustc_trans/trans/base.rs

```diff
@@ -623,7 +623,7 @@ pub fn compare_simd_types<'blk, 'tcx>(
                     size: uint,
                     op: ast::BinOp)
                     -> ValueRef {
-    match t.sty {
+    let cmp = match t.sty {
```

Comment:
Nice! I did something similar in my branch, though I had all the matching in another function and only needed the signedness from this one.